frontend/packages/topology/src/filters/FilterProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+import { getDefaultAppliedFilters } from '../redux/reducer';
+import { DEFAULT_TOPOLOGY_FILTERS } from './const';
+import { getAppliedFilters } from '../redux/action';
+import { DisplayFilters } from '../topology-types';
+
+const TOPOLOGY_DISPLAY_FILTERS_CONFIG_MAP_KEY = `devcosole.topology.filters`;
+
+const getTopologyFilters = (appliedFilters: AppliedFilters) => {
+  const filters = [...DEFAULT_TOPOLOGY_FILTERS];
+  filters.forEach((filter) => {
+    if (appliedFilters[filter.id] !== undefined) {
+      filter.value = appliedFilters[filter.id];
+    }
+  });
+
+  return filters;
+};
+
+type AppliedFilters = { [filterKey: string]: boolean };
+type SetTopologyFilters = (filters: DisplayFilters) => void;
+
+const useFilterContextValues = (): [
+  DisplayFilters,
+  AppliedFilters,
+  boolean,
+  SetTopologyFilters,
+] => {
+  const [appliedFilters, setAppliedFilters, appliedFiltersLoaded] = useUserSettings(
```

Comment:
@invincibleJai Should this not use the fallback wrapper?

Reply:
Thanks for noticing this, used the wrapper `useUserSettingsCompatibility`.